Always update CsvReader.ColumnCount

Rob-Hague opened the pull request:
Fixes #1779

Rand-Random commented:
maybe the test should also validate if columncount decreases on rows with fewer columns, as to check that columcount is a max(columcount)

Rob-Hague commented:
> maybe the test should also validate if columncount decreases on rows with fewer columns, as to check that columcount is a max(columcount)

Thanks. I've strengthened the test, but actually that is not the current behaviour when `DetectColumnCountChanges` is true: the test passes on master when `DetectColumnCountChanges` is true (i.e. `ColumnCount` always changes) but fails when it is false (`ColumnCount` always returns 0). With the PR the test passes for both cases.

I also removed my misleading comment on `columnCount` and renamed it to `prevColumnCount`.
